### PR TITLE
Removed unused property hints and `Object::get_translatable_strings()`

### LINKED ISSUE #30203: Review and cleanup GlobalScope constants for 4.0
**Godot version:**
Master (e2bbf2cba)

**Issue description:**
The `@GlobalScope` documentation lists a number of constants which are no longer used, especially among property hints, usage flags and method flags.

The list should be reviewed for 4.0 to drop obsolete stuff, including their duplicate entries in the GDNative API.

See also #16863.

### PR DESCRIPTION
**Pull request by reduz**
* Remove unused `EditorPropertyMember` and related hints, previouly used by
  VisualScript. Such logic should be implemented in the VS module itself.
* As the above broke compatibility with the VS module, clean up the other
  hacks that were still in core in support of VisualScript.
* `PROPERTY_USAGE_INTERNATIONALIZED` was only used in Object's
  `get_translatable_strings()`, which is a legacy function not used anywhere.
  So both are removed.
* Reordered some usage flags after the above removal to minimize the diff.
* General clean up.

Fixes #30203.

Co-authored-by: @akien-mga

### CODE REVIEW

File: core/object/object.h [hunk 1]
```diff
@@ -130,7 +122,6 @@ enum PropertyUsageFlags {
 	PROPERTY_USAGE_ARRAY = 1 << 28, // Used in the inspector to group properties as elements of an array.
 
 	PROPERTY_USAGE_DEFAULT = PROPERTY_USAGE_STORAGE | PROPERTY_USAGE_EDITOR,
-	PROPERTY_USAGE_DEFAULT_INTL = PROPERTY_USAGE_DEFAULT | PROPERTY_USAGE_INTERNATIONALIZED,
```
[the remark this answers is not in the document]
Comment (akien-mga) [reply core/object/object.h:133]: It was only used 3 times, and was the only place where `PROPERTY_USAGE_INTERNATIONALIZED` was used. That didn't seem like a use case that warrants having its own convenience combined usage flag.

Looking further into it, it seems `PROPERTY_USAGE_INTERNATIONALIZED` might actually be useless, so we might be able to remove both. It's only used in `Object::get_translatable_strings()` which is never used.

[the remark this answers is not in the document]
Comment (KoBeWi) [reply core/object/object.h:133]: I think `get_translatable_strings()` is supposed to be used by POT generator. Not sure why it isn't.

[the remark this answers is not in the document]
Comment (akien-mga) [reply core/object/object.h:133]: That's what I thought at first, but it doesn't seem to be the case AFAICT. It's not used in the engine, and not exposed for plugins. I think we came up with a separate system of parsers for the POT generator.

**Review by YuriSizov — APPROVED** [review]

File: editor/editor_themes.cpp [hunk 2]
```diff
@@ -133,25 +133,6 @@ void EditorColorMap::create() {
...
-	add_conversion_color_pair("#b9ec41", "#96ce1a"); // VisualScript Transform2D
-	add_conversion_color_pair("#f74949", "#f77070"); // VisualScript Plane
-	add_conversion_color_pair("#ec418e", "#ec69a3"); // VisualScript Quat
-	add_conversion_color_pair("#ee5677", "#ee7991"); // VisualScript AABB
-	add_conversion_color_pair("#e1ec41", "#b2bb19"); // VisualScript Basis
-	add_conversion_color_pair("#f68f45", "#f49047"); // VisualScript Transform
-	add_conversion_color_pair("#417aec", "#6993ec"); // VisualScript NodePath
-	add_conversion_color_pair("#41ec80", "#2ce573"); // VisualScript RID
-	add_conversion_color_pair("#55f3e3", "#12d5c3"); // VisualScript Object
-	add_conversion_color_pair("#54ed9e", "#57e99f"); // VisualScript Dictionary
```
[the remark this answers is not in the document]
Comment (YuriSizov) [reply editor/editor_themes.cpp:154]: Yes, there is a lot of pollination in these things. We can merge and fix those icons afterwards

[the remark this answers is not in the document]
Comment (akien-mga) [reply editor/editor_themes.cpp:154]: I readded those, and clarified that they're Variant type icons, not VisualScript only.

**Comment by akien-mga** [comment]
@raulsntos Thanks, included.

**Comment by akien-mga** [comment]
Thanks!

File: core/object/object.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (YeldhamDev) [reply core/object/object.h:133]: If I remember correctly, the parsers themselves take care of string extraction, also allowing new extractors to be created.

**Pull request merged**